Xamarin.PropertyEditing.Mac/Themes/MacThemeManager.cs

```diff
@@ -26,6 +28,24 @@ protected override void SetTheme ()
 					CurrentAppearance = NSAppearance.CurrentAppearance;
 					break;
 			}
+
+			this.themeCache.Clear ();
+		}
+
+		public string GetImageNameForTheme (string imageNamed, string selectedSuffix = "")
+		{
+			return (Theme == PropertyEditorTheme.Dark ? imageNamed + "~dark" : imageNamed) + selectedSuffix;
+		}
+
+		public NSImage GetImageForTheme (string imageNamed, string selectedSuffix = "")
+		{
+			var imageNameForTheme = GetImageNameForTheme (imageNamed, selectedSuffix);
+
+			if (!this.themeCache.TryGetValue (imageNameForTheme, out NSImage themeImage)) {
```

Comment:
I'm only just discovering how good these inline declarations are and weirdly though the scope would only be within the if. But since it is an `out`, it make sense.